lib/ansible/module_utils/aws/direct_connect.py

```diff
@@ -0,0 +1,62 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+"""
+This module adds shared support for Direct Connect modules.
+"""
+
+import traceback
+try:
+    import botocore
+except ImportError:
+    pass
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+
+class DirectConnectError(Exception):
```

Comment:
Thanks for reviewing! The reason I made a custom exception class was to be able to set an appropriate error message wherever the error took place since I'm trying to avoid having to pass module into each function just so I can call fail_json/fail_json_aws directly. I could not add try/excepts anywhere but main() as well, but I'll lose the ability to pass along a helpful message of what operation went wrong (ex: https://github.com/ansible/ansible/pull/26152/files#diff-0a1d9c2db7323d55147e2fa3ef2556b4R154). If you prefer a different solution I'm happy to make changes.

Your right about leveraging fail_json_aws now we have it. I will simplify my logic here https://github.com/ansible/ansible/pull/26152/files#diff-0a1d9c2db7323d55147e2fa3ef2556b4R304 to use it.